deploy/awstools/awstools.py

```diff
@@ -78,21 +86,22 @@ def aws_resource_names():
         instanceid = res.stdout
         rootLogger.debug(instanceid)
 
-        resp = client.describe_tags(
-            Filters=[
+        operation_params = {
+            'Filters': [
                 {
                     'Name': 'resource-id',
                     'Values': [
                         instanceid,
                     ]
                 },
             ]
-        )
+        }
+        resp_pairs = depaginated_boto_query(client, 'describe_tags', operation_params, 'Tags')
     except client.exceptions.ClientError:
         return base_dict
 
     resptags = {}
-    for pair in resp['Tags']:
+    for pair in resp_pairs:
         resptags[pair['Key']] = pair['Value']
     rootLogger.debug(resptags)
 
```

Comment:
@sagark , I thought the thing that was failing in the tutorial was the `describe_images` call below here in `get_f1_ami_id`.  I really dislike how GH tries to keep me from reading and commenting on code that is too far from one of your changes but I'm pretty sure that needs to be updated too, right?

Even if we're fairly confident that the server-side filtering should prevent pagenation, we should still probably handle it.

Reply:
Separate from functionality, I think it also makes sense to change any client call to the paginated version anyways.

Reply:
The problem at the tutorial was with `describe_fpga_images`.

The `describe_images` API doesn't use pagination (note that there's no `NextToken` or `MaxResults` param like the other calls have):
https://boto3.amazonaws.com/v1/documentation/api/1.20.21/reference/services/ec2.html?highlight=describe_images#EC2.Client.describe_images

To double check, I tried to create a paginator for describe_images and I get an `Operation cannot be paginated` error.